src/solc_0.8.13/raffle/GenericRaffle.sol

```diff
@@ -0,0 +1,407 @@
+        sandOwner = _owner;
+        emit SandOwnerSet(_owner);
+    }
+
+    function _baseURI() internal view virtual override returns (string memory) {
+        return baseTokenURI;
+    }
+
+    function setBaseURI(string memory baseURI) public onlyOwner {
+        require(bytes(baseURI).length != 0, "baseURI is not set");
+        baseTokenURI = baseURI;
+        emit BaseURISet(baseURI);
+    }
+
+    function setSignAddress(address _signAddress) external onlyOwner {
+        require(_signAddress != address(0x0), "Sign address is zero address");
+        signAddress = _signAddress;
+        emit SignAddressSet(_signAddress);
+    }
+
+    function _msgSender()
+        internal
+        view
+        override(ContextUpgradeable, ERC2771HandlerUpgradeable)
+        returns (address sender)
+    {
+        return ERC2771HandlerUpgradeable._msgSender();
+    }
+
+    function _msgData() internal view override(ContextUpgradeable, ERC2771HandlerUpgradeable) returns (bytes calldata) {
+        return ERC2771HandlerUpgradeable._msgData();
+    }
+
+    function renounceOwnership() public virtual override onlyOwner {
+        revert("Renounce ownership is not available");
+    }
+
+    // Empty storage space in contracts for future enhancements
+    // ref: https://github.com/OpenZeppelin/openzeppelin-contracts-upgradeable/issues/13)
+    uint256[50] private __gap;
+
+    function setApprovalForAll(address operator, bool approved)
+        public
+        override(ERC721Upgradeable, IERC721Upgradeable)
+        onlyAllowedOperatorApproval(operator)
+    {
+        super.setApprovalForAll(operator, approved);
+    }
+
+    function approve(address operator, uint256 tokenId)
+        public
+        override(ERC721Upgradeable, IERC721Upgradeable)
+        onlyAllowedOperatorApproval(operator)
+    {
+        super.approve(operator, tokenId);
+    }
+
+    function transferFrom(
+        address from,
+        address to,
+        uint256 tokenId
+    ) public override(ERC721Upgradeable, IERC721Upgradeable) onlyAllowedOperator(from) {
+        super.transferFrom(from, to, tokenId);
+    }
+
+    function safeTransferFrom(
+        address from,
+        address to,
+        uint256 tokenId
+    ) public override(ERC721Upgradeable, IERC721Upgradeable) onlyAllowedOperator(from) {
+        super.safeTransferFrom(from, to, tokenId);
+    }
+
+    function safeTransferFrom(
+        address from,
+        address to,
+        uint256 tokenId,
+        bytes memory data
+    ) public override(ERC721Upgradeable, IERC721Upgradeable) onlyAllowedOperator(from) {
+        super.safeTransferFrom(from, to, tokenId, data);
+    }
+
+    /**
+     * @dev assume the contract has an owner, but leave specific Ownable implementation up to inheriting contract
+     */
+    function owner() public view override(OwnableUpgradeable, UpdatableOperatorFiltererUpgradeable) returns (address) {
+        return OwnableUpgradeable.owner();
+    }
+}
```

Comment:
Functions are very randomly ordered, should be organised well to increase readability

Reply:
also, the functions are missing the natspec